src/dftbp/dftbplus/main.F90

```diff
@@ -845,21 +847,29 @@ subroutine processOutputCharges(env, this)
     #:endif
 
       if (.not. this%tRealHS) then
-        ! denseSubtractDensityOfAtoms_cmplx_periodic()
-        call denseSubtractDensityOfAtoms(env, this%ints, this%denseDesc, this%parallelKS,&
-            & this%neighbourList, this%kPoint, this%nNeighbourSK, this%iCellVec, this%cellVec,&
-            & this%iSparseStart, this%img2CentCell, this%q0, this%densityMatrix%deltaRhoOutCplx)
-        ! Build real-space delta rho from k-space density matrix:
-        ! Also, zero-out deltaRhoOutCplxHS, because of internal summation.
-        this%densityMatrix%deltaRhoOutCplxHS(:,:,:,:,:,:) = 0.0_dp
-        call transformDualSpaceToBvKRealSpace(this%densityMatrix%deltaRhoOutCplx,&
-            & this%parallelKS, this%kPoint, this%kWeight, this%hybridXc%bvKShifts,&
-            & this%densityMatrix%deltaRhoOutCplxHS)
-      #:if WITH_MPI
-        ! Distribute all square, real-space density matrices to all nodes via global summation
-        call mpifx_allreduceip(env%mpi%interGroupComm, this%densityMatrix%deltaRhoOutCplxHS,&
-            & MPI_SUM)
-      #:endif
+        if (this%hybridXc%hybridXcAlg == hybridXcAlgo%matrixBased) then
+          ! denseSubtractDensityOfAtoms_cmplx_periodic_global()
+          call denseSubtractDensityOfAtoms(env, this%ints, this%denseDesc, this%neighbourList,&
+              & this%kPoint, this%densityMatrix%iKiSToiGlobalKS, this%nNeighbourSK, this%iCellVec,&
+              & this%cellVec, this%iSparseStart, this%img2CentCell, this%q0,&
+              & this%densityMatrix%deltaRhoOutCplx)
+        else
+          ! denseSubtractDensityOfAtoms_cmplx_periodic()
```

Comment:
Any reason to keep the commented line?

Reply:
Above every call of the `denseSubtractDensityOfAtoms` interface I indicated which routine is actually invoked. It was really helpful for me to not get confused all the time. Shall I remove all of those comments?